app/models/report.rb

```diff
@@ -14,4 +24,37 @@ def editable?(target_user)
   def created_on
     created_at.to_date
   end
+
+  def execute_save_procedure
+    transaction do
+      save!
+      refresh_mentions
+    end
+    true
+  rescue ActiveRecord::RecordInvalid
+    false
```

Comment:
参考：今回はこのままでもいいですが、例外の発生を前提としたロジックは上から読めないコードになる（問題が起きたらrescueへジャンプするGOTO文ロジックになる）ので、個人的にはなるべく例外に頼らないロジック構成にするのがいいと思っています。

```suggestion
    success = true
    transaction do
      success &= save
      success &= refresh_mentions
      raise ActiveRecord::Rollback unless success
    end
    success
```